artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -177,9 +177,9 @@
 
    private final ScheduledDeliveryHandler scheduledDeliveryHandler;
 
-   private long messagesAdded;
+   private AtomicLong messagesAdded = new AtomicLong(0);
```

Comment:
To be more memory friendly where lots of queues can the long be volatile and then use a AtomicLongFieldUpdater to safely update.